pkg/infra/state_reader/state_converter/pulumi.go

```diff
@@ -0,0 +1,78 @@
+package stateconverter
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/klothoplatform/klotho/pkg/construct"
+	statetemplate "github.com/klothoplatform/klotho/pkg/infra/state_reader/state_template"
+	"go.uber.org/zap"
+)
+
+type (
+	PulumiState []Resource
+
+	Resource struct {
+		Urn     string                 `json:"urn"`
+		Type    string                 `json:"type"`
+		Outputs map[string]interface{} `json:"outputs"`
+	}
+	pulumiStateConverter struct {
+		templates map[string]statetemplate.StateTemplate
+	}
+)
+
+func (p pulumiStateConverter) ConvertState(data []byte) (State, error) {
+	var pulumiState PulumiState
+	err := json.Unmarshal(data, &pulumiState)
+	if err != nil {
+		return nil, err
+	}
+	internalModel := make(State)
+	var errs error
+	// Convert the Pulumi state to the internal model
+	for _, resource := range pulumiState {
+		mapping, ok := p.templates[resource.Type]
+		if !ok {
+			zap.S().Warnf("no mapping found for resource type %s", resource.Type)
```

Comment:
i guess we dont know if its not needed, i was thinking warning because my eventualy goal would be to error, but i guess well never support all resources. Still feels like its something for the user to be aware of

Reply:
Hm, maybe I misunderstood how this works then. I thought the input graph would be used for which things connect to other things, then the reading the state would fill out the properties (like id). In that scenario, I'd expect there to be an error if a resource in the input graph doesn't have a template. Sounds like that's not how that works. Can you explain what the input graph is for then? I could probably figure it out by looking closer at the code, but if you could give a quick description that'd help.